tests/osaka/eip7934_block_rlp_limit/test_max_block_rlp_size.py

```diff
@@ -0,0 +1,341 @@
+                            transactions.append(adjusted_tx)
+                            break
+
+                        adjusted_diff = abs(target_size - adjusted_test_size)
+                        if adjusted_diff < best_diff:
+                            best_diff = adjusted_diff
+                else:
+                    raise RuntimeError(
+                        "Failed to find a transaction that matches the target size."
+                    )
+        else:
+            transactions.append(empty_tx)
+
+    final_size = get_block_rlp_size(
+        transactions, gas_used=sum(tx.gas_limit for tx in transactions)
+    )
+    final_gas = sum(tx.gas_limit for tx in transactions)
+
+    assert final_size == target_size, f"Size mismatch: got {final_size}, expected {target_size}"
+    return transactions, final_gas
+
+
+@pytest.mark.parametrize(
+    "from_limit",
+    [
+        -1,  # max RLP size - 1 byte, valid
+        0,  # exactly max RLP size, valid
+        pytest.param(  # max RLP size + 1 byte, invalid
+            1,
+            marks=pytest.mark.exception_test,
+        ),
+    ],
+)
+def test_block_at_rlp_size_limit_boundary(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+    post: Alloc,
+    block_size_limit: int,
+    env: Environment,
+    exact_size_transactions,
+    from_limit: int,
+):
+    """
+    Test the block rlp size limit.
+
+    - At the limit - 1 byte, the block is valid
+    - At the limit, the block is valid
+    - At the limit + 1 byte, the block is invalid
+    """
+    transactions, gas_used = exact_size_transactions
+    block_rlp_size = get_block_rlp_size(transactions, gas_used=gas_used)
+    assert block_rlp_size == block_size_limit, (
+        f"Block RLP size {block_rlp_size} does not exactly match limit {block_size_limit}, "
+        f"difference: {block_rlp_size - block_size_limit} bytes"
+    )
+
+    block = Block(
+        txs=transactions,
+        exception=BlockException.RLP_BLOCK_LIMIT_EXCEEDED if from_limit == 1 else None,
+    )
+
+    if from_limit == -1:
+        # remove last byte
+        block.extra_data = Bytes(EXTRA_DATA_AT_LIMIT[:-1])
+    elif from_limit == 0:
+        block.extra_data = Bytes(EXTRA_DATA_AT_LIMIT)
+    elif from_limit == 1:
+        block.extra_data = Bytes(EXTRA_DATA_AT_LIMIT + b"\x00")
+    else:
+        raise ValueError(f"Invalid `from_limit` value: {from_limit}")
+    block.timestamp = ZeroPaddedHexNumber(HEADER_TIMESTAMP)
+    blockchain_test(
+        genesis_environment=env,
+        pre=pre,
+        post=post,
+        blocks=[block],
+        verify_sync=False if from_limit == 1 else True,
+    )
+
+
+@pytest.mark.exception_test
+def test_block_slightly_over_rlp_size_limit(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+    post: Alloc,
+    env: Environment,
+    exact_size_transactions,
+    block_size_limit: int,
+):
+    """Test that a block at exactly the MAX_RLP_BLOCK_SIZE is accepted."""
```

Comment:
This looks like it is from an older test case we removed. We should update this. 

We could probably parametrize more test cases in the test above instead? Since we cache the rlp-limit block now, we can probably afford to check as many cases as we want, -2, -3, +2, +3 etc?